src/Products/SiteErrorLog/tests/testInitialization.py

```diff
@@ -0,0 +1,83 @@
+##############################################################################
+#
+# Copyright (c) 2001, 2002 Zope Foundation and Contributors.
+# All Rights Reserved.
+#
+# This software is subject to the provisions of the Zope Public License,
+# Version 2.1 (ZPL).  A copy of the ZPL should accompany this distribution.
+# THIS SOFTWARE IS PROVIDED "AS IS" AND ANY AND ALL EXPRESS OR IMPLIED
+# WARRANTIES ARE DISCLAIMED, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF TITLE, MERCHANTABILITY, AGAINST INFRINGEMENT, AND FITNESS
+# FOR A PARTICULAR PURPOSE.
+#
+##############################################################################
+
+import os
+import shutil
+import tempfile
+import unittest
+
+from App.config import getConfiguration, setConfiguration
+from OFS.Application import Application, AppInitializer
+from Zope2.Startup.options import ZopeWSGIOptions
+
+good_cfg = """
+instancehome <<INSTANCE_HOME>>
+
+<zodb_db main>
+   mount-point /
+   <mappingstorage>
+      name mappingstorage
+   </mappingstorage>
+</zodb_db>
+"""
+
+original_config = None
+
+
+def getApp():
+    from App.ZApplication import ZApplicationWrapper
+    DB = getConfiguration().dbtab.getDatabase('/')
+    return ZApplicationWrapper(DB, 'Application', Application)()
+
+
+class TestInitialization(unittest.TestCase):
+    """ Test the application initialization """
+
+    def setUp(self):
+        global original_config
+        if original_config is None:
+            original_config = getConfiguration()
+        self.TEMPNAME = tempfile.mkdtemp()
+
+    def tearDown(self):
+        import App.config
+        App.config.setConfiguration(original_config)
+        shutil.rmtree(self.TEMPNAME)
+        import Products
+        Products.__path__ = [d for d in Products.__path__
+                             if os.path.exists(d)]
+
+    def configure(self, text):
+        # We have to create a directory of our own since the existence
+        # of the directory is checked.  This handles this in a
+        # platform-independent way.
+        config_path = os.path.join(self.TEMPNAME, 'zope.conf')
+        with open(config_path, 'w') as fd:
+            fd.write(text.replace(u"<<INSTANCE_HOME>>", self.TEMPNAME))
+
+        options = ZopeWSGIOptions(config_path)()
+        config = options.configroot
+        self.assertEqual(config.instancehome, self.TEMPNAME)
+        setConfiguration(config)
+
+    def getOne(self):
```

Comment:
Why is this method named `getOne`?
It seems to return the Zope initializer. Maybe it could be named `getInitializer`.